seahorn/jobs/hash_table_swap/aws_hash_table_swap_harness.c

```diff
@@ -0,0 +1,42 @@
+#include <aws/common/hash_table.h>
+#include <seahorn/seahorn.h>
+#include <hash_table_helper.h>
+#include <utils.h>
+
+int main(void)
+{
+    struct aws_hash_table a;
+    struct aws_hash_table b;
+    bool inita = nd_bool();
+    bool initb = nd_bool();
+    struct store_byte_from_buffer stored_byte_a;
+    struct store_byte_from_buffer stored_byte_b;
+
+    /* There are no loops in the code under test, so use the biggest possible value */
+    if (inita)
+    {
+        initialize_bounded_aws_hash_table(&a, SIZE_MAX);
+        assume(aws_hash_table_is_valid(&a));
+        save_byte_from_hash_table(&a, &stored_byte_a);
+    }
+
+    if (initb)
+    {
+        initialize_bounded_aws_hash_table(&b, SIZE_MAX);
+        assume(aws_hash_table_is_valid(&b));
+        save_byte_from_hash_table(&b, &stored_byte_b);
+    }
+    aws_hash_table_swap(&a, &b);
```

Comment:
Note that non det `inita` and `initb` are carried over from cbmc jobs;
if `a` and/or `b` are not initialized, there will definitely be read and copy of undefined value in `aws_hash_table_swap`;
and this is caught by seahorn with the below warning:
```
Warning: found 6 possible reads of undefined values                                                                                                                                                                           
Possible read of undefined value at                                                                                                                                                                                           
--- File   : aws-c-common/source/hash_table.c                                                                                                                                                                                 
--- Line   : 271                                                                                                                                                                                                              
--- Column : 10                                                                                                                                                                                                               
--- Bitcode:   %244 = phi %struct.hash_table_state* [ %157, %save_byte_from_hash_table.exit52.i ], [ undef, %124 ], !dbg !1083
```

Reply:
remove `inita` and `initb`. Always initialize the structures. Otherwise there is undefined behaviour and verification result is meaningless. 

That's my only comment. The rest looks good.

Reply:
thanks, not sure why CBMC folks decided to do their verification with non-det initialized data structured in this case; maybe different semantics again

Reply:
for them it is non-deterministically initialized structures. But I am not sure why this is any better than initialized with non-deterministic values. The code just copies fields over from one struct to another. 

You can also initialize them with `memhavoc` explicitly instead